Jenkinsfile

```diff
@@ -441,6 +443,43 @@ def nodeWithCleanup(String label, Closure steps) {
   }
 }
 
+def runSimulator(String appPath, String bundleId, String arguments) {
+  def id = UUID.randomUUID().toString().replace('-', '')
+  try {
+    def runtimes = sh returnStdout: true, script: 'xcrun simctl list devicetypes runtimes'
+
+    def runtimeId;
+
+    def runtimeMatcher = (runtimes =~ /iOS.*\((?<runtimeId>com.apple.CoreSimulator.SimRuntime.iOS[^\)]*)\)/)
```

Comment:
@fealebenpae suggested that we move the regex matching to `sh` as well. I wasn't able to figure out quote escaping, but the shell script that does it should look something like:
```bash
runtimes=`xcrun simctl list runtimes`
regex="\((com.apple.CoreSimulator.SimRuntime.iOS[^\)]*)\)"
if [[ $runtimes =~ $regex ]]
then
  runtimeId="${BASH_REMATCH[1]}"
else
  echo "Failed to find iOS runtime."
  exit 1
fi
```